packages/shopify-codemod/transforms/console-utils.js

```diff
@@ -0,0 +1,68 @@
+import chalk from 'chalk';
+
+const NEWLINE_REGEX = /\r\n|[\n\v\f\r\x85\u2028\u2029]/;
+const CONTEXT_LINES = 2;
+
+function padLeft(text, length) {
+  return (' '.repeat(length) + text).slice(-length);
+}
+
+function paintRange(lineNum, startLine, startColumn, endLine, endColumn) {
+  let paintStart = Number.MAX_SAFE_INTEGER;
+  let paintEnd = Number.MAX_SAFE_INTEGER;
+
+  if (startLine <= lineNum && lineNum <= endLine) {
+    if (lineNum === startLine) {
+      paintStart = startColumn;
+    } else {
+      paintStart = 0;
+    }
+    if (lineNum === endLine) {
+      paintEnd = endColumn;
+    } else {
+      paintEnd = Number.MAX_SAFE_INTEGER;
+    }
+  }
+  return {paintStart, paintEnd};
+}
+
+function paintLine(line, from, to, color) {
+  return `${line.slice(0, from)}${color(line.slice(from, to))}${line.slice(to)}`;
+}
+
+function formatLine(line, lineNum) {
+  return `${chalk.gray(padLeft(lineNum + 1, 4))}|${line}\n`;
+}
+
+function codeContext(node, source, color) {
```

Comment:
You can even split the "get context lines" and "add highlight info" into separate functions, but it gets a bit verbose.